diff --git a/Makefile_sysfw-image b/Makefile_sysfw-image
index 08d80c35..8e489bf1 100644
--- a/Makefile_sysfw-image
+++ b/Makefile_sysfw-image
@@ -13,6 +13,7 @@ SYSFW_SOC_am65xx-hs-evm = am65x
 SYSFW_SOC_am64xx-evm = am64x
 SYSFW_SOC_am64xx-hs-evm = am64x
 SYSFW_SOC_am62xx-evm = am62x
+SYSFW_SOC_am62xx-rs10 = am62x
 SYSFW_SOC_j7-evm = j721e
 SYSFW_SOC_j7-hs-evm = j721e
 SYSFW_SOC = $(SYSFW_SOC_$(PLATFORM))
