diff --git a/plat/imx/imx8m/imx8mm/platform.mk b/plat/imx/imx8m/imx8mm/platform.mk
index bfbcaed38..dc2cb2240 100644
--- a/plat/imx/imx8m/imx8mm/platform.mk
+++ b/plat/imx/imx8m/imx8mm/platform.mk
@@ -147,7 +147,7 @@ ERRATA_A53_835769	:=	1
 ERRATA_A53_843419	:=	1
 ERRATA_A53_855873	:=	1
 
-BL32_BASE		?=	0xbe000000
+BL32_BASE		?=	0x7e000000
 $(eval $(call add_define,BL32_BASE))
 
 BL32_SIZE		?=	0x2000000
