diff --git a/opkg-build b/opkg-build
index 5dbf903..4ba9a05 100755
--- a/opkg-build
+++ b/opkg-build
@@ -131,8 +131,8 @@ You probably want to chown these to a system user: " >&2
 	disallowed_filename=`disallowed_field Filename`
 	[ "$?" -ne 0 ] && PKG_ERROR=1
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "*** Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])" >&2
+	if echo $pkg | grep '[^a-zA-Z0-9._+-]'; then
+		echo "*** Error: Package name $name contains illegal characters, (other than [a-zA-Z0-9._+-])" >&2
 		PKG_ERROR=1;
 	fi
 
