diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index 9ab3e1e95133e852899233b2f3b4fb2e8fcb60d9..aa8b123a90c7b35bcf0ec9114dd93ef160bb7364 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -71,7 +71,8 @@ mx8mn-flavorlist = \
 
 mx8mp-flavorlist = \
 	mx8mpevk \
-	mx8mp_rsb3720_6g
+	mx8mp_rsb3720_6g \
+	mx8mprsb3720a2 \
 
 mx8qm-flavorlist = \
 	mx8qmmek \
@@ -433,6 +434,13 @@ $(call force,CFG_CORE_LARGE_PHYS_ADDR,y)
 $(call force,CFG_CORE_ARM64_PA_BITS,36)
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mprsb3720a2))
+CFG_DDR_SIZE ?= UL(0x180000000)
+CFG_UART_BASE ?= UART3_BASE
+$(call force,CFG_CORE_LARGE_PHYS_ADDR,y)
+$(call force,CFG_CORE_ARM64_PA_BITS,36)
+endif
+
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mp_rsb3720_6g))
 CFG_DDR_SIZE ?= UL(0x180000000)
 CFG_UART_BASE ?= UART3_BASE
diff --git a/scripts/nxp_build.sh b/scripts/nxp_build.sh
index f596a53c21e0160bf15a91125cb8154eb30655db..caaa2b4fdb9c9d4b4700bf4621e399a3d6bb98ae 100755
--- a/scripts/nxp_build.sh
+++ b/scripts/nxp_build.sh
@@ -37,6 +37,7 @@ boards_list=(
 	imx-mx8mmevk \
 	imx-mx8mnevk \
 	imx-mx8mpevk \
+	imx-mx8mprsb3720a2 \
 	imx-mx8qxpmek \
 	imx-mx8qmmek \
 	imx-mx8qmmekcockpita53 \
