diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index b4fbfed5..d7421a8d 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -17,6 +17,7 @@ mx6q-flavorlist = \
 	mx6qsabresd \
 	mx6qhmbedge \
 	mx6qapalis \
+	mx6qrsb4411a1 \
 
 mx6qp-flavorlist = \
 	mx6qpsabreauto \
@@ -268,6 +269,11 @@ CFG_DDR_SIZE ?= 0x40000000
 CFG_NS_ENTRY_ADDR ?= 0x12000000
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6qrsb4411a1))
+CFG_DDR_SIZE ?= 0x40000000
+CFG_NS_ENTRY_ADDR ?= 0x12000000
+endif
+
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx6qpsabreauto mx6qsabreauto \
 	mx6dlsabreauto mx6solosabreauto))
 CFG_DDR_SIZE ?= 0x80000000
diff --git a/scripts/nxp_build.sh b/scripts/nxp_build.sh
index 9085a883..f534c84a 100755
--- a/scripts/nxp_build.sh
+++ b/scripts/nxp_build.sh
@@ -7,6 +7,7 @@ boards_list=(
 	mx6dlhmbedge \
 	mx6qsabrelite \
 	mx6qsabresd \
+	mx6qrsb4411a1 \
 	mx6qsabreauto
 	mx6qhmbedge \
 	mx6qpsabresd \
