diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index c190750..a510564 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -69,7 +69,8 @@ mx8mn-flavorlist = \
 	mx8mnevk
 
 mx8mp-flavorlist = \
-	mx8mpevk
+	mx8mpevk \
+	mx8mpepcr5710a1 \
 
 mx8qm-flavorlist = \
 	mx8qmmek \
@@ -347,6 +348,13 @@ CFG_DDR_SIZE ?= 0x80000000
 CFG_UART_BASE ?= UART2_BASE
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mpepcr5710a1))
+CFG_DDR_SIZE ?= 0x180000000ULL
+CFG_UART_BASE ?= UART3_BASE
+$(call force,CFG_CORE_LARGE_PHYS_ADDR,y)
+$(call force,CFG_CORE_ARM64_PA_BITS,36)
+endif
+
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mpevk))
 CFG_DDR_SIZE ?= 0x180000000ULL
 CFG_UART_BASE ?= UART2_BASE
diff --git a/scripts/nxp_build.sh b/scripts/nxp_build.sh
index 3819893..a275aef 100755
--- a/scripts/nxp_build.sh
+++ b/scripts/nxp_build.sh
@@ -34,6 +34,7 @@ boards_list=(
 	mx8mmevk \
 	mx8mnevk \
 	mx8mpevk \
+	mx8mpepcr5710a1 \
 	mx8qxpmek \
 	mx8qmmek \
 	mx8dxlevk \
